training/src/main/java/com/wolox/training/models/User.java

```diff
@@ -45,7 +44,8 @@ public User() {
     this.books = new ArrayList<>();
   }
 
-  @ManyToMany(cascade = {CascadeType.MERGE, CascadeType.REFRESH})
+  @JsonIgnore
+  @ManyToMany(cascade = {CascadeType.MERGE, CascadeType.REFRESH}, fetch = FetchType.EAGER)
```

Comment:
Instead of using EAGER, what you should do to avoid the lazy initialization is use the @Transactional annotation in the service layer. This way, the database transaction won't be commited/closed until all the method ends. 
So, as a conclusion: you can use @Transactional from Spring in the service methods which persist information